publish/api.py

```diff
@@ -34,14 +34,26 @@ def publish(self, request, *args, **kwargs):
         data = serializer.validated_data
         return Response(publish_dandiset.delay(**data).id)
 
-    @action(detail=True, methods=['GET'], url_path='(?P<version>[^/.]+)')
+    @action(detail=True, methods=['GET'], url_path='(?P<version>[.0-9]+)')
     def version(self, request, dandi_id, version):
         queryset = self.filter_queryset(self.get_queryset())
         filter_kwargs = {'dandi_id': dandi_id, 'version': version}
         obj = get_object_or_404(queryset, **filter_kwargs)
         serializer = DandisetSerializer(obj)
         return Response(serializer.data)
 
+    @action(detail=True, methods=['GET'], url_path='versions')
+    def versions(self, request, dandi_id):
+        queryset = self.filter_queryset(self.get_queryset())
+        filter_kwargs = {'dandi_id': dandi_id}
+        get_object_or_404(queryset, **filter_kwargs)
+        versions = [v['version'] for v in
+                    Dandiset.objects
+                    .filter(**filter_kwargs)
+                    .values('version')
```

Comment:
Trying to limit the columns returned with `.values` seems like premature optimization. I think we should only break out of the ORM (i.e. return dicts instead of objects) when it's shown to be necessary for performance.